File: src/main/java/lesson2/Group.java
package lesson2;

class Group {
    private int ind = -1;
    private Employee[] employees = new Employee[10];
    private String name;

    public Group(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        String employeesStr = "";
        for (int i = 0; i <= ind; ++i) {
            employeesStr += "\n" + employees[i].toString();
        }
        return "Group{" +
                "name='" + name + '\'' +
                "\nemployees=[" + employeesStr + "\n]" +
                "}";
    }

    public void add(Employee e) {
        if (ind == employees.length - 1) {
            throw new RuntimeException("Нет места для нового сотрудника");
        }
        ind++;
        employees[ind] = e;
    }

    public void clearAll() {
        ind = -1;
        employees = new Employee[10];
    }

    public void remove(int i) {
        if (i > ind) {
            return;
        }
        employees[i] = null;
        for (int j = i; j <= ind; ++j) {
            if (j + 1 < employees.length) {
                employees[j] = employees[j + 1];
            }
        }
        --ind;
    }
}

File: src/main/kotlin/lesson2_kt/Group.kt
package lesson2_kt

//1.Создайте класс Сотрудник, с полями: имя, email, возраст, должность. Сотрудник должен уметь отпечатать в консоль информацию о себе;
//2.Создайте класс Группа, включающий в себя название группы и массив из максимум 10 сотрудников (не все элементы массива могут быть заполнены).
// Реализуйте возможность добавлять сотрудников в этот массив, удалять их из него по индексу, и удалять всех разом;
//3.В классе Группа должен быть метод, позволяющий отпечатать информацию обо всех сотрудниках, входящих в эту группу;

data class Group (
    var name : String
) {
    private var index = -1
    private var array = arrayOfNulls<Employee>(10)

    fun add(employee: Employee) {
        if (index == array.size-1) {
            throw RuntimeException("В массиве нет места")
        }
        index++
        array[index] = employee
    }

    fun clear() {
        index = -1
        array = arrayOfNulls<Employee>(10)
    }

    fun remove(ind : Int) {
        if (ind > index) {
            return
        }
        array[ind] = null
        for (i in ind..index) {
            if (i+1 == array.size)
                break
            array[i] = array[i+1]
        }
        --index
    }

    override fun toString(): String {
        var sb = StringBuilder()
        for (item in array) {
            if (item != null) {
                sb.append(item.toString() + "\n")
            }

        }
        return "Group(name='$name', array=$sb)"
    }


}